extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/AuthRuntimeConfig.java

```diff
@@ -39,11 +40,20 @@ public class AuthRuntimeConfig {
     public Map<String, List<String>> rolesMapping;
 
     /**
-     * Properties file containing the client certificate common name (CN) to role mappings.
+     * Client certificate attribute whose values are going to be mapped to the 'SecurityIdentity' roles
+     * according to the roles mapping specified in the certificate properties file.
+     * The attribute must be either one of the Relative Distinguished Names (RDNs) or Subject Alternative Names (SANs).
+     * By default, the Common Name (CN) attribute value is used for roles mapping.
+     */
+    @ConfigItem(defaultValue = "DN_CN")
+    public CertificateRoleAttribute certificateRoleAttribute;
```

Comment:
IMHO it should be a String, in case someone will want to refer to a custom attribute

Reply:
> IMHO it should be a String, in case someone will want to refer to a custom attribute

we cannot really support reliably string for SAN, we can do it for custom DNs if you would like. I am not sure what the name should be as we will have 2 config properties and attribute is not correct name

Reply:
@sberyozkin there is more than one way to specify SAN general name and I am not sure I want to write parser

Reply:
@michalvavrik I see, can you at least leave only those ones which might be used as mappers, and we can keep adding more based on some real cases.

Reply:
truth be told @sberyozkin I originally had string but I didn't figure how to name this as I'd need 2 config properties and I'd prefer SAN to be enum and DN to be string. Maybe we could make it string with some values expected (SAN_xy) and some parsed DN_whatever? I really don't know.

Reply:
thanks for feedback, let me try something in between...